packages/rum-core/src/domain/vital/vitalCollection.ts

```diff
@@ -9,61 +9,91 @@ import { PageState } from '../contexts/pageStateHistory'
   name: string
-  startClocks: ClocksState
   context?: Context
+  details?: string
 }
 
 export interface DurationVitalStop {
+  context?: Context
+  details?: string
+}
+
+export interface DurationVitalAdd {
   name: string
-  stopClocks: ClocksState
+  startClocks: ClocksState
+  duration: Duration
   context?: Context
+  details?: string
+}
+
+export interface DurationVitalInstance {
+  stop: (options?: DurationVitalStop) => void
 }
 
 interface DurationVital {
   name: string
   type: VitalType.DURATION
   startClocks: ClocksState
-  value: Duration
+  duration: Duration
+  details?: string
   context?: Context
 }
 
 export function startVitalCollection(lifeCycle: LifeCycle, pageStateHistory: PageStateHistory) {
-  const vitalStartsByName = new Map<string, DurationVitalStart>()
+  function isValid(vital: DurationVital) {
+    return !pageStateHistory.wasInPageStateDuringPeriod(PageState.FROZEN, vital.startClocks.relative, vital.duration)
+  }
 
-  lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
-    // Discard all the vitals that have not been stopped to avoid memory leaks
-    vitalStartsByName.clear()
-  })
+  function addDurationVital(vitalAdd: DurationVitalAdd) {
+    const vital = assign({ type: VitalType.DURATION }, vitalAdd)
 
-  function isValid(vital: DurationVital) {
-    return !pageStateHistory.wasInPageStateDuringPeriod(PageState.FROZEN, vital.startClocks.relative, vital.value)
+    if (isValid(vital)) {
+      lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processVital(vital, true))
+    }
   }
 
   return {
-    startDurationVital: (vitalStart: DurationVitalStart) => {
-      vitalStartsByName.set(vitalStart.name, vitalStart)
-    },
-    stopDurationVital: (vitalStop: DurationVitalStop) => {
-      const vitalStart = vitalStartsByName.get(vitalStop.name)
-      if (!vitalStart) {
+    addDurationVital,
+    startDurationVital: (startVital: DurationVitalStart) =>
+      createVitalInstance((vital) => {
+        addDurationVital(vital)
+      }, startVital),
+  }
+}
+
+export function createVitalInstance(
+  stopCallback: (vital: DurationVital) => void,
+  vitalStart: DurationVitalStart
+): DurationVitalInstance {
+  const startClocks = clocksNow()
+  let stopClocks: ClocksState | undefined
+
+  return {
+    stop: (vitalStop) => {
+      if (stopClocks) {
         return
       }
-      const vital = buildDurationVital(vitalStart, vitalStop)
-      vitalStartsByName.delete(vital.name)
-      if (isValid(vital)) {
-        lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processVital(vital, true))
-      }
+
+      stopClocks = clocksNow()
+
+      stopCallback(buildDurationVital(vitalStart, startClocks, vitalStop, stopClocks))
     },
   }
 }
 
-function buildDurationVital(vitalStart: DurationVitalStart, vitalStop: DurationVitalStop) {
+function buildDurationVital(
+  vitalStart: DurationVitalStart,
+  startClocks: ClocksState,
+  stopOptions: DurationVitalStop = {},
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
  vitalStop: DurationVitalStop = {},
```